spec/view_component/storybook/dsl/controls_dsl_spec.rb

```diff
@@ -142,4 +142,43 @@
                        }
     end
   end
+
+  describe "#custom" do
+    subject do
+      custom(first_name: "J.R.R.", last_name: "Tolkien") do |first_name:, last_name:|
+        Author.new(first_name: first_name, last_name: last_name).full_name
+      end.param(:author)
+    end
+
+    include_examples "has controls attributes",
+                     {
+                       class: ViewComponent::Storybook::Controls::CustomConfig,
+                       param: :author,
+                       name: "Author"
+                     }
+
+    it "returns cutom value from params" do
+      from_params = subject.value_from_params({})
+      expect(from_params).to eq("J.R.R. Tolkien")
+    end
+  end
+
+  describe "#klazz" do
+    subject do
+      klazz(Author, first_name: "J.R.R.", last_name: "Tolkien").param(:author)
```

Comment:
Seeing the `.param(author)` so late in the line seemed a bit off to me... potentially a rearrangement might look like `klazz(:author, Author, ...`.

I got a bit tripped up by the syntax the first time reading it. Or maybe we just call it `custom_param(:author, Author, ...`?

Reply:
Ignore me. I'm realizing that this DSL is for the spec. I looked back at #50 again to better understand, and I like the klazz shorthand now. :)

Reply:
Yeah. Honestly there is very little reason I can think of that a consumer would actually need to call `param` themselves. We infer it from the name of args anyway. Maybe if you're using `*args` or `**kwargs` then you'd want to set it? But even then that actual over the wire params is mostly an internal detail. It's more likely you want to override the name.

For these specs the only reason I'm including the param is so we can a param and a name to test against. Even then maybe I should drop it from these specs as its confusing?